packages/core/src/utils/mixin-builder.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Class} from './common-types';
+
+// tslint:disable:no-any
+
+/**
+ * Interface for functions that can mix properties/methods into a base class
+ *
+ * For example,
+ * ```
+ * var calculatorMixin = Base => class extends Base {
+ *   calc() { }
+ * };
+ *
+ * function timestampMixin(Base) {
+ *   return class extends Base {
+ *     created: Date = new Date();
+ *     modified: Date = new Date();
+ *   }
+ * }
+ * ```
+ * See http://justinfagnani.com/2015/12/21/real-mixins-with-javascript-classes/.
+ */
+export interface MixinFunc {
+  <BC extends Class<any>>(baseClass: BC): BC;
+}
+
+// /**
```

Comment:
Unfortunately I am not able to figure out a way to keep the fluent api signature, nor the class MixinBuilder. 

The reason is it's not possible to store the `baseClass` as a property of class, because declaring it with type `Class<any>` causes the extended class doesn't preserve the return type of base class method.